Refactor code for Load Effects

BigRoy opened the pull request:
## Changelog Description

Restructure code in Load Effects

## Additional review information

Happened to just be reading over this file and felt like it was in need of cleanup.

- re-use code between load and update
- byteify is not needed in Py3+ (as far as I know no more hassle with unicodes)
- no need for `OrderedDict` because `dict` is guaranteed to be insertion ordered in Py3.7+ (note: does this suffice?)
    - Nuke 13 was the first release with Py3, which seems to be Py3.7.7?
- cosmetics: no camelcase var names
- use `self.filepath_from_context` in `update()`

Similar cleanup should likely happen to **LoadEffectsInputProcess**. It has a similar issue of `load()` and `update()` basically containing the same long code. And even better 99% of the loader is the same as this loader. I feel like we may even be better off putting both loaders in one file and just re-use logic where we can between them.

## Testing notes:

1. Load Effects should work
2. Update of Load Effects should work, etc.

jakubjezek001 reviewed (changes requested):
```
During load error happened on Product: "effectPlateP01" Representation: "effectJson" Version: 2

Error message: Unsupported type: <class 'NoneType'>

Traceback (most recent call last):
  File "C:\CODE\__YNPUT\ayon-core\client\ayon_core\tools\loader\models\actions.py", line 740, in _load_representations_by_loader
    load_with_repre_context(
  File "C:\CODE\__YNPUT\ayon-core\client\ayon_core\pipeline\load\utils.py", line 325, in load_with_repre_context
    return loader.load(repre_context, name, namespace, options)
  File "C:\CODE\__YNPUT\ayon-nuke\client\ayon_nuke\plugins\load\load_effects.py", line 66, in load
    return containerise(
  File "C:\CODE\__YNPUT\ayon-nuke\client\ayon_nuke\api\pipeline.py", line 433, in containerise
    set_avalon_knob_data(node, data)
  File "C:\CODE\__YNPUT\ayon-nuke\client\ayon_nuke\api\lib.py", line 116, in wrapper
    return decorated_func(*args, **kwargs)
  File "C:\CODE\__YNPUT\ayon-nuke\client\ayon_nuke\api\lib.py", line 486, in set_avalon_knob_data
    imprint(node, create, tab=tab_name)
  File "C:\CODE\__YNPUT\ayon-nuke\client\ayon_nuke\api\lib.py", line 395, in imprint
    for knob in create_knobs(data, tab):
  File "C:\CODE\__YNPUT\ayon-nuke\client\ayon_nuke\api\lib.py", line 330, in create_knobs
    knobs += create_knobs(value)
  File "C:\CODE\__YNPUT\ayon-nuke\client\ayon_nuke\api\lib.py", line 336, in create_knobs
    raise TypeError("Unsupported type: %r" % type(value))
TypeError: Unsupported type: <class 'NoneType'>

```

I was testing it from latest develop and it is not happening.

jakubjezek001 commented:
@BigRoy is this waiting for the fix?

BigRoy commented:
So - this was me just in nightly hours looking at the code and seeing it and thinking "that looks ancient" and refactored it. The issue with 'fixing' it for me is mostly that:

1. I do not have access to Nuke.
2. I haven't used Nuke (so even if I did have access it may take a while to exactly figure out what's up.

Having said that, since your last test I did push a commit that would log the 'errorring' knob (and it should also ignore it). So it should 'work' now but we should still investigate the new log then about which knob now failed that didn't before.

Any chance you could re-run and get the log output?

I noticed that Load Gizmo files are also basically 90% the same code as the Load Effects. I really have a feeling we can somehow merge that logic so that the remainder is solely a few lines of code for each individual loader because there isn't actually that much unique about each of them. The _actual_ load logic is very small - there's just a lot of stuff being done around it.

jakubjezek001 commented:
here is the updated log. 

```
Traceback (most recent call last):
  File "C:\CODE\__YNPUT\ayon-core\client\ayon_core\tools\loader\models\actions.py", line 740, in _load_representations_by_loader
    load_with_repre_context(
  File "C:\CODE\__YNPUT\ayon-core\client\ayon_core\pipeline\load\utils.py", line 325, in load_with_repre_context
    return loader.load(repre_context, name, namespace, options)
  File "C:\CODE\__YNPUT\ayon-nuke\client\ayon_nuke\plugins\load\load_effects.py", line 66, in load
    return containerise(
  File "C:\CODE\__YNPUT\ayon-nuke\client\ayon_nuke\api\pipeline.py", line 433, in containerise
    set_avalon_knob_data(node, data)
  File "C:\CODE\__YNPUT\ayon-nuke\client\ayon_nuke\api\lib.py", line 116, in wrapper
    return decorated_func(*args, **kwargs)
  File "C:\CODE\__YNPUT\ayon-nuke\client\ayon_nuke\api\lib.py", line 487, in set_avalon_knob_data
    imprint(node, create, tab=tab_name)
  File "C:\CODE\__YNPUT\ayon-nuke\client\ayon_nuke\api\lib.py", line 396, in imprint
    for knob in create_knobs(data, tab):
  File "C:\CODE\__YNPUT\ayon-nuke\client\ayon_nuke\api\lib.py", line 330, in create_knobs
    knobs += create_knobs(value)
  File "C:\CODE\__YNPUT\ayon-nuke\client\ayon_nuke\api\lib.py", line 336, in create_knobs
    raise TypeError(
TypeError: Unsupported type for key: '('avalon:namespace', 'namespace')': <class 'NoneType'>
```

jakubjezek001 commented:
> I really have a feeling we can somehow merge that logic so that the remainder is solely a few lines of code for each individual loader because there isn't actually that much unique about each of them

yeah, lets merge it and clear this up a little.

BigRoy commented:
@jakubjezek001 I think I have fixed the `namespace` knob issue - it should now match previous logic. I've also refactored so that logic is also shared with Load Gizmo. However, I did it quite quickly and may be somewhat messy.

I'm expecting it to not work :D but since I can't test it myself... please bear with me and let me know what you find.

jakubjezek001 commented:
hey @BigRoy sorry for the delay, could we please resolve the conflicts so I can merge it?
